test_spawn_broken_pipe unit test is flaky

This [failed the last two Debian mips64el builds](https://buildd.debian.org/status/fetch.php?pkg=cockpit&arch=mips64el&ver=303-1&stamp=1697698026&file=log) and also occasionally [happens on GitHub](https://github.com/cockpit-project/cockpit/actions/runs/6794869078/job/18471982872?pr=19583#step:5:304) -- I've seen it twice just this morning.

```
___________________________ test_spawn_broken_pipe ____________________________

bridge = <test.pytest.test_peer.Bridge object at 0x7fd9ac539990>

    @pytest.mark.asyncio
    async def test_spawn_broken_pipe(bridge):
[...]
        peer = BrokenPipePeer(specific_error=False)
        with pytest.raises(BrokenPipeError):
>           await peer.start()

test/pytest/test_peer.py:204: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    async def start(self, init_host: Optional[str] = None, **kwargs: JsonDocument) -> JsonObject:
[...]

        try:
            # Wait for something to happen:
            #   - exception from our connection function
            #   - receiving "init" from the other side
            #   - receiving EOF from the other side
            #   - .close() was called
            #   - other transport exception
>           init_message = await self.init_future
E           cockpit.peer.PeerExited: 9

/tmp/.tox/py37-pytest/lib/python3.7/site-packages/cockpit/peer.py:100: PeerExited
=========================== short test summary info ============================
FAILED test/pytest/test_peer.py::test_spawn_broken_pipe - cockpit.peer.PeerExited: 9
```

I didn't succeed with getting this failure locally with unmodified code, even with two parallel loops running this 100 times.

With this patch below it fails reliably:
```diff
--- test/pytest/test_peer.py
+++ test/pytest/test_peer.py
@@ -189,6 +189,7 @@ async def test_spawn_broken_pipe(bridge):
         async def do_connect_transport(self) -> None:
             transport = await self.spawn(['sh', '-c', 'exit 9'], ())
             assert isinstance(transport, SubprocessTransport)
+            await asyncio.sleep(0.5)
             # The process will exit soon — try writing to it until a write fails.
             while not transport.is_closing():
                 transport.write(b'x')
```

That might also give a hint what's wrong: The asyncio main loop apparently runs in between the self.spawn() and the first .write(), and sometimes the process is quick enough to already finish in between that?

--- a/test/pytest/test_peer.py
+++ b/test/pytest/test_peer.py
@@ -187,8 +187,10 @@ async def test_spawn_broken_pipe(bridge):
             self.specific_error = specific_error
 
         async def do_connect_transport(self) -> None:
-            transport = await self.spawn(['sh', '-c', 'exit 9'], ())
+            transport = await self.spawn(['sh', '-c', 'read a; exit 9'], ())
             assert isinstance(transport, SubprocessTransport)
+            # Make the process exit by writing a newline (causing `read` to finish)
+            transport.write(b'\n')
             # The process will exit soon — try writing to it until a write fails.
             while not transport.is_closing():
                 transport.write(b'x')

